tools/ttysnoop.py

```diff
@@ -162,6 +162,22 @@ def usage():
 #endif
 """
 
+kernel_release = open('/proc/sys/kernel/osrelease').read().split('.')
+if int(kernel_release[0]) >= 5 and int(kernel_release[1]) >= 8:
+    PERF_MODE = "USE_BPF_RING_BUF"
+else:
+    PERF_MODE = "USE_BPF_PERF_BUF"
```

Comment:
This is unreliable due to backport, currently, you can use `kernel_struct_has_field()` API.

Or maybe introduce a new API `kernel_has_struct()`.

Reply:
Replace with "BPF.kernel_struct_has_field(b'bpf_ringbuf', b'waitq') == 1" works.
One more question, it's my first time to pull request, there are already 2 commits, it will be 3 after I upload a refine patch. I was going to upload one patch for version fix, and another for ringbuf, how could I do? Will it be ok if I just upload later refine patch, or need I revert the ringbuf patch and create a new branch to upload?

Reply:
Let us do two separate commits: one for version fix and the other is ringbuf fix.

Reply:
Hi，ringbuf patch is reverted